core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -681,10 +684,10 @@ private void updateTransientDependencies() {
         @Override
         protected URL findResource(String name) {
             if (PluginManager.FAST_LOOKUP) {
-                    for (PluginWrapper pw : getTransitiveDependencies()) {
-                        URL url = ClassLoaderReflectionToolkit._findResource(pw.classLoader, name);
-                        if (url!=null)    return url;
-                    }
+                for (PluginWrapper pw : getTransitiveDependencies()) {
+                    URL url = ClassLoaderReflectionToolkit._findResource(pw.classLoader, name);;
```

Comment:
```suggestion
                    URL url = ClassLoaderReflectionToolkit._findResource(pw.classLoader, name);
```

and suggest reverting entire hunk anyway